### PR TITLE
adding company fields to createAccount, updateAccount, getAccount

### PR DESCRIPTION
**Pull request by stephanwampouille**
> Doing this blind without BE changes, so might need adjustments based on @makk94 feedback. Will keep In draft state until

Have seen Marina's comments and made changes, still a question for Mahina to answer and to review my CLI code (first time altering it and want to be sure)

### CODE REVIEW

File: javascript/sdk/lib/controllers/iam.ts [hunk 1]
```diff
@@ -48,22 +48,30 @@ export default class IAMController {
...
     return true;
   }
 
   /** Get account properties */
-  async getAccount(options: RequestOptions = {}) {
-    const response = await this.#client.requestWithAuth("/iam/account", {
-      method: "GET",
-      ...options,
-    });
+  async getAccount(company: string = "", options: RequestOptions = {}) {
```
Comment (makk94) [inline javascript/sdk/lib/controllers/iam.ts:66]: no new inputs on getAccount. Just showing the company in the response now

File: javascript/sdk/lib/controllers/iam.ts [hunk 2]
```diff
@@ -48,22 +48,30 @@ export default class IAMController {
   }
 
   /** Update properties on an account */
-  async updateAccount(mode: Mode, options: RequestOptions = {}) {
+  async updateAccount(
```
Comment (makk94) [inline javascript/sdk/lib/controllers/iam.ts:51]: like the other update functions, I'm gonna return the account in the response now

**Comment by makk94** [comment]
will you update the CLI too?

File: javascript/sdk/lib/controllers/iam.ts [hunk 3]
```diff
@@ -17,12 +17,12 @@ export default class IAMController {
   }
 
   async newAccount(
-    { email, name }: CreateAccountParams,
+    { email, name, company }: CreateAccountParams,
```
Comment (stephanwampouille) [inline javascript/sdk/lib/controllers/iam.ts:20]: should I default company to `""` @makk94  or will BE take care of this and return is as an empty string if no value is passed?

**Comment by stephanwampouille** [comment]
> will you update the CLI too?

done, but can you double check please. First time I have touched CLI code and want to make sure I dont mess anything up. @makk94

**Comment by stephanwampouille** [comment]
@ceci21 you should be aware of CLI changes here

**Review by privateducky — APPROVED** [review]

File: javascript/sdk/lib/controllers/iam.ts [hunk 4]
```diff
@@ -48,14 +48,18 @@ export default class IAMController {
...
+  ) {
+    const response = await this.#client.requestWithAuth("/iam/account", {
       method: "PUT",
-      body: JSON.stringify({ mode }),
+      body: JSON.stringify({ mode, company }),
       ...options,
     });
 
-    return true;
+    return (await response.json()) as Account;
```
[the remark this answers is not in the document]
Comment (makk94) [reply javascript/sdk/lib/controllers/iam.ts:62]:
```
  flat = dict(
        account_id=account.account_id,
        whoami=auth['user'],
        company=account.company,
        mode=account.mode,
        controls=[ct['name'] for ct in account.controls],
        users=list(account.users.values()),
        queue=account.queue
    )
```
is what we're returning

BE changes are merged but not in us2 yet, so probably explains your tests

File: python/sdk/prelude_sdk/controllers/iam_controller.py [hunk 5]
```diff
@@ -10,10 +10,14 @@ def __init__(self, account):
         self.account = account
 
     @verify_credentials
-    def new_account(self, user_email: str, user_name: str):
+    def new_account(self, user_email: str, user_name: str, company: str = None):
+        body = dict(handle=user_email, user_name=user_name)
```
Comment (makk94) [inline python/sdk/prelude_sdk/controllers/iam_controller.py:14]: user_name and company are both optional - it'd be nice to be consistent (i'd go with giving user name an `if user_name:` clause like you do with company)

File: python/sdk/prelude_sdk/controllers/iam_controller.py [hunk 6]
```diff
@@ -40,16 +44,21 @@ def purge_account(self):
         return res.text
 
     @verify_credentials
-    def update_account(self, mode: int):
+    def update_account(self, mode: int, company: str = None):
```
Comment (makk94) [inline python/sdk/prelude_sdk/controllers/iam_controller.py:47]: same here, both mode and company are optional, so set mode in an if block too

File: python/cli/prelude_cli/views/iam.py [hunk 7]
```diff
@@ -34,12 +35,20 @@ def register_account(controller):
...
+              type=str)
 @click.pass_obj
 @handle_api_error
-def update_account(controller, mode):
+def update_account(controller, mode, company):
     """ Update an account """
     with Spinner():
-        controller.update_account(mode=Mode[mode.upper()].value)
+        data = controller.update_account(
+            mode=Mode[mode.upper()].value,
```
Comment (makk94) [inline python/cli/prelude_cli/views/iam.py:48]: watch out that both these flags are optional. you'll want to change the mode flag default to None and do an `if` check around it before sending it to the controller

we didn't used to need ^ when it the only thing you could update was mode, but now someone might want to update company without updating mode

**Pull request merged**